loginchecker.php

```diff
@@ -0,0 +1,135 @@
+ * CLI it conforms to the Nagios plugin standard.
+ *
+ * See also:
+ *  - http://nagios.sourceforge.net/docs/3_0/pluginapi.html
+ *  - https://nagios-plugins.org/doc/guidelines.html#PLUGOUTPUT
+ *
+ */
+
+defined('MOODLE_INTERNAL');
+
+define('NO_UPGRADE_CHECK', true);
+
+$options = array(
+    'help' => false,
+    'critthresh' => 500,
+    'warnthresh' => 10,
+    'logtime' => 5,
+);
+
+if (isset($argv)) {
+    // If run from the CLI.
+    define('CLI_SCRIPT', true);
+    require_once(__DIR__ . '/../../../config.php');
+    require_once('nagios.php');
+    require_once($CFG->libdir . '/clilib.php');
+
+    list($options, $unrecognized) = cli_get_params($options,
+    array(
+        'h' => 'help',
+        )
+    );
+
+    $checktime = time() - ($options['logtime'] * 60);
+
+    if ($unrecognized) {
+        $unrecognized = implode("\n  ", $unrecognized);
+        cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+    }
+
+    if ($options['help']) {
+        print "Checks the moodle database for failed login attempts in a given time frame
+
+        loginchecker.php [options]
+
+        Options:
+        -h, --help            Print out this help
+            --critthresh=n    Threshold for number of failed logins to trigger a critical error (default 500 attempts)
+            --warnthresh=n    Threshold for number of failed logins to trigger a warning (default 10 attempts)
+            --logtime=n       Time in minutes to check back for a critical error (default 5 minutes prior)
+
+        Example:
+        \$sudo -u www-data /usr/bin/php admin/tool/heartbeat/loginchecker.php --logtime=60\n";
+
+        die;
+    }
+
+} else {
+    // If run from the web.
+    define('NO_MOODLE_COOKIES', true);
+    require_once(__DIR__ . '/../../../config.php');
+    require_once('nagios.php');
+
+    $options['critthresh'] = (optional_param('critthresh', null, PARAM_INT) != null ? 
+        optional_param('critthresh', null, PARAM_INT) : 500);
+    $options['warnthresh'] = (optional_param('warnthresh', null, PARAM_INT) != null ? 
+        optional_param('warnthresh', null, PARAM_INT) : 10);
+    $options['logtime'] = (optional_param('logtime', null, PARAM_INT) != null ? 
+        optional_param('logtime', null, PARAM_INT) : 5);
+
+    $checktime = time() - ($options['logtime'] * 60);
+
+    header("Content-Type: text/plain");
+
+    // Make sure varnish doesn't cache this. But it still might so go check it!
+    header('Pragma: no-cache');
+    header('Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate, proxy-revalidate');
+    header('Expires: Tue, 04 Sep 2012 05:32:29 GMT');
+}
+
+// Check that the database doesn't need to be upgraded.
+if (moodle_needs_upgrading()) {
+    send_critical("Moodle upgrade pending, login checker execution suspended");
+}
+
+$sqlstring = "SELECT count(*) AS logincount, other, origin, ip FROM mdl_logstore_standard_log WHERE target = 'user_login'
```

Comment:
please format this sql

https://docs.moodle.org/dev/SQL_coding_style#Indentation

also never use mdl_xyz also wasy do {xyz}

Also make $checktime an sql param never inline arguments

Reply:
Also didn't I previously mention removing the 'GROUP BY other, origin, ip' ? We only care about the total number